src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultPlugin.cs

```diff
@@ -166,7 +166,7 @@ internal static byte[] GenerateInitializationVector()
 
                 var encryptor = aesAlg.CreateEncryptor(aesAlg.Key, aesAlg.IV);
 
-                return await PerformCryptography(encryptor, payload);
+                return await PerformCryptography(encryptor, payload).ConfigureAwait(false);
```

Comment:
i dont think that is the case. you can only avoid a `.ConfigureAwait(false);` when u can direct return the task. in this case u cannot since u need the dispose to happen after the task completes.